src/main/java/org/opensearch/neuralsearch/search/query/HybridCollectorManager.java

```diff
@@ -46,6 +52,7 @@ public abstract class HybridCollectorManager implements CollectorManager<Collect
     private final boolean isSingleShard;
     private final int trackTotalHitsUpTo;
     private final SortAndFormats sortAndFormats;
+    private final Optional<Weight> filterWeight;
```

Comment:
if you want to stick with Optional approach add Optional suffix to the variable name